src/raft.c

```diff
@@ -408,31 +432,47 @@ void RaftExecuteCommandArray(RedisRaftCtx *rr,
         /* Explanation:
          * When we have an ACL, we will have a user set on the context, so need "C"
          */
-        char *resp_call_fmt = cmds->acl ? "CE0v" : "E0v";
+        char *resp_call_fmt;
+        if (cmds->cmd_flags & CMD_SPEC_MULTI) {
+            /* don't block inside a MULTI */
+            resp_call_fmt = cmds->acl ? "CE0v" : "E0v";
+        } else {
+            resp_call_fmt = cmds->acl ? "KCE0v" : "KE0v";
+        }
 
         enterRedisModuleCall();
         RedisModule_SetContextUser(ctx, user);
         reply = RedisModule_Call(ctx, cmd, resp_call_fmt, &c->argv[1], c->argc - 1);
+        RedisModule_SetContextUser(ctx, NULL);
```

Comment:
Why do we need this?

Reply:
after executing the commnd, we want to reset state to be safe.